MetaMask/metamask-mobile
# fix: legacy gas miss match and not preserved after change

## **Description**

Whenever I trigger a legacy transaction from a dapp, I can see how the Estimated gas fees does not match the gas details (if I open the Edit gas). Furthermore if I change the suggested gas, I can see how the value is not changed and tx is submitted with the suggested gas value instead of the one I set. [See here](https://github.com/MetaMask/metamask-mobile/pull/7035#issuecomment-1717118939)

_Write a short description of the changes included in this pull request, also include relevant motivation and context. Have in mind the following questions:_
_1. What is the reason for the change?_
_2. What is the improvement/solution?_

## **Manual testing steps**
Test for both 1559 supported network and legacy network. For legacy network, use ganache and check that `--hardfork berlin` or use Palm. For 1559, a testnet like Georli should suffice

Setup legacy network using 
_1. Step1:_ Setup wallet and import the right key and account
_3. Step3:_ Check that transactions are submitted as expected.

## **Screenshots/Recordings**

### **Before**

https://github.com/MetaMask/metamask-mobile/assets/29962968/b816bfb0-fe78-4b86-81a4-0e554cfd38cc


https://github.com/MetaMask/metamask-mobile/assets/29962968/c7d7b889-2fac-4560-8896-e1b330b42f13



### **After**
Send token: https://recordit.co/KCujgfnwOO
Send legacy: http://recordit.co/QOSHpcmrMS
Create a token: http://recordit.co/ygpx2T3Wsw
Token approve: http://recordit.co/C4ItxYxWCB
Send legacy txn from Georli: http://recordit.co/TUmaOp9vHl

## **Related issues**

_Fixes #7290 

Bitrise build: https://app.bitrise.io/app/be69d4368ee7e86d/pipelines/35e839a1-6d20-491d-bfea-6956c207776a

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
